### PR TITLE
Set k8sgateway api version when setting up e2e test cluster

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description

When setting up our kind cluster for tests, we need to ensure that the correct version of the k8sgateway APIs are loaded, as we support multiple versions of Istio that use different versions of the rapidly changing k8sgateway APIs.

To support this, `k8sgateway_api_version` has been added to the the min/max environment variables for the test matrix and the value are set as the `CONFORMANCE_VERSION` environment variable when running the setup-kind script.

## CI changes
The setup kind cluster action now has an optional parameter `k8sgateway_api_version` which is set as the  `CONFORMANCE_VERSION` environment variable when running the setup-kind script.

`k8sgateway_api_version` has been set in the `min/max_versions.env` files

# Context
Istio was failing to install when running the "min" tests because the expected CRDs were not present

## Testing steps
Nightly test run without istio install failures: https://github.com/solo-io/gloo/actions/runs/11955940423

specifically `istioctl install failed` is not found in the "min" test log: https://github.com/solo-io/gloo/actions/runs/11955940423/job/33329519257

## Interesting decsions
We should bump the minimum version of Istio, but will leave that to a separate/wholistic effort to determine minimum versions for the new release instead of cherry-picking this spot.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by andy-fong — APPROVED** [review]

**Comment by jenshu** [comment]
@sheidkamp can you sync with @danehans on https://github.com/k8sgateway/k8sgateway/pull/10360
probably makes sense to have the same implementation in both repos

File: .github/workflows/composite-actions/setup-kind-cluster/action.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ inputs:
   istio-version:
     required: true
     description: The version of Istio
+  k8sgateway_api_version:
```
Comment (sam-heilbron) [inline .github/workflows/composite-actions/setup-kind-cluster/action.yaml:24]:
```suggestion
  k8sgateway-api-version:
```

The existing inputs follow this convention. I can't find the github docs now, but I had tried to base it off them. If I was wrong and your convention is preferred, can we change the other inputs in this file so we're consistent?

Comment (sheidkamp) [reply .github/workflows/composite-actions/setup-kind-cluster/action.yaml]: Updated in the appropriate places. Left the underscores in the env files as that is the usual formatting for env variables, and the switch in conventions comes when the env variable is assigned to the action input.

**Review by sam-heilbron — APPROVED** [review]
nice work!

**Comment by danehans** [comment]
xref https://github.com/solo-io/gloo/pull/10390 is a downstream cherry-pick of https://github.com/k8sgateway/k8sgateway/pull/10360 for supporting Gateway API matrix testing.

**Pull request merged**